code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/PayloadItem.java

```diff
@@ -26,9 +28,9 @@ class PayloadItem implements Serializable {
     final ItemData data;
 
     PayloadItem(final Map<String, Object> payloadItemMap) {
-        id = (String) payloadItemMap.get(ID);
-        schema = (String) payloadItemMap.get(SCHEMA);
-        final Map<String, String> dataMap = (Map<String, String>) payloadItemMap.get(DATA);
-        data = new ItemData(dataMap.get(ID), dataMap.get(CONTENT));
+        id = DataReader.optString(payloadItemMap, ID, "");
+        schema = DataReader.optString(payloadItemMap, SCHEMA, "");
```

Comment:
these default values are communicating that values `id` and `schema` are optional for `PayloadItem`.  we don't want that behavior, right?  should we throw with an `InvalidInitException` if required properties are not available?

Reply:
yup that sounds like the correct behavior. will update for PayloadItem and PropositionInfo.